ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -98,9 +117,40 @@ - (void)viewDidLoad
 {
   [super viewDidLoad];
 
+  // EXKernel.appRegistry.homeAppRecord does not contain any homeAppRecord until this point,
+  // therefore we cannot move this propoerty initialization to the constructor/initializer
+  _isHomeApp = _appRecord == [EXKernel sharedInstance].appRegistry.homeAppRecord;
+  
+  // show LoadingCancelView in managed apps only
+  if (!self.isStandalone && !self.isHomeApp) {
+    self.appLoadingCancelView = [EXAppLoadingCancelView new];
+    // if home app is available then LoadingCancelView can show `go to home` button
+    if ([EXKernel sharedInstance].appRegistry.homeAppRecord) {
+      self.appLoadingCancelView.delegate = self;
+    }
+    [self.view addSubview:self.appLoadingCancelView];
+    [self.view bringSubviewToFront:self.appLoadingCancelView];
+  }
+  
+  // show LoadingProgressWindow in managed apps and dev home app only
+  BOOL isDevelopmentHomeApp = self.isHomeApp && [EXEnvironment sharedEnvironment].isDebugXCodeScheme;
+  self.appLoadingProgressWindowController = [[EXAppLoadingProgressWindowController alloc] initWithEnabled:!self.isStandalone || isDevelopmentHomeApp];
+
+  // show SplashScreen in standalone apps and home app only
+  // SplashScreen for managed is shown once the manifest is available
+  EXSplashScreenService *splashScreenService = (EXSplashScreenService *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXSplashScreenService class]];
+  if (self.isHomeApp) {
+    self.homeAppSplashScreenViewProvider = [EXHomeAppSplashScreenViewProvider new];
+    [splashScreenService showSplashScreenFor:self
+                    splashScreenViewProvider:self.homeAppSplashScreenViewProvider
+                             successCallback:^{}
+                             failureCallback:^(NSString *message){ UMLogWarn(@"%@", message); }];
+  } else if (self.isStandalone) {
+    [splashScreenService showSplashScreenFor:self];
```

Comment:
Nitpick, but it would be better for this clause to be before the `self.isHomeApp` — it's kind of lost and harder to notice after the previous long block with this different indentation